src/Nethermind/Nethermind.Consensus.AuRa/AuRaBlockProducer.cs

```diff
@@ -96,5 +101,21 @@ protected override bool PreparedBlockCanBeMined(Block block)
             
             return false;
         }
+
+        protected override Block ProcessPreparedBlock(Block block)
+        {
+            var processedBlock = base.ProcessPreparedBlock(block);
+            
+            // We need to check if we are withing gas limit. We cannot calculate this in advance because:
+            // a) GasLimit can come from contract
+            // b) Some transactions that call contracts can be added to block and we don't know how much gas they will use.
+            if (processedBlock.GasUsed > processedBlock.GasLimit)
+            {
+                if (Logger.IsError) Logger.Error($"Block produced used {processedBlock.GasUsed} gas and exceeded gas limit {processedBlock.GasLimit}.");
```

Comment:
this sems like being at risk of getting stuck and not producing any more blocks - we need to ensure that some block cn be produced?

Reply:
I will discuss it again with POA, but quite sure this is how it should be, as contract is our source of truth here that we trust. If its broken than yes node will not produce